src/libertem/io/dataset/base/tiling.py

```diff
@@ -540,9 +540,15 @@ def get_scheme(
         full_base_shape = (1,) + tuple(base_shape)
         min_factors = (depth,) + tuple(min_factors)
 
+        if roi is None:
+            containing_shape = partition.shape
+        else:
+            roi_slice = partition.slice.get(roi.reshape(-1), nav_only=True)
+            containing_shape = (np.count_nonzero(roi_slice), ) + tuple(partition.shape.sig)
```

Comment:
Could also use `partition.slice.adjust_for_roi(roi).shape` for the same effect, I think

Reply:
Thx, that's shorter! :-)